xapian-core/tests/api_wrdb.cc

```diff
@@ -1589,6 +1589,50 @@ DEFINE_TESTCASE(nomoredocids1, writable) {
     return true;
 }
 
+DEFINE_TESTCASE(synonym_merge1, writable && synonyms) {
+    Xapian::WritableDatabase db = get_writable_database();
+    Xapian::WritableDatabase db1 = get_named_writable_database("db1");
+    Xapian::WritableDatabase db2 = get_named_writable_database("db3");
+    Xapian::WritableDatabase db3 = get_named_writable_database("db4");
+
+    db.add_synonym("abc", "asd");
+    db.add_synonym("abc", "asd1");
+
+    db1.add_synonym("abc", "asd2");
+    db1.add_synonym("abc", "asd3");
+
+    db2.add_synonym("abc", "asd4");
+    db2.add_synonym("abc", "asd5");
+    db2.add_synonym("abc", "asd6");
+
+    db3.add_synonym("abc", "asd7");
+    db3.add_synonym("abc", "asd8");
+    db3.add_synonym("abc", "asd9");
+    db3.add_synonym("abc", "asd10");
```

Comment:
What exactly are you trying to test here?

In general a comment before each testcase explaining what the testcase is actually for is useful (a lot of older testcases lack this, but most newer ones include it).  I can see it's testing merging of synonyms across subdbs, but I'm not really sure about why in detail.

The synonyms from each subdb mostly form disjoint ranges (except that `asd10` string sorts between `asd1` and `asd2`) which seems like a lost opportunity to shake out problems in the merging code.  At least for the testcase as it is, it seems it would provide equally good coverage with just 2 sub databases.

Reply:
Yep! I will change to be just two dbs.